### PR TITLE
Add a function that recomputes the stats based on votes (not tested)

### PR DESCRIPTION
**Pull request by Ked57**

### CODE REVIEW

File: functions/src/update-stats.ts [hunk 1]
```diff
@@ -21,39 +21,40 @@ export const updateStats = functions.firestore
...
+    const storedVote = await transaction.get(
+      statsCollection
+        .doc(vote.campaign)
+        .collection("votes")
+        .doc(voteId)
+    );
+
+    if (storedVote.exists) {
+      console.info(
+        "This vote(" + voteId + ") has already been counted, aborting;"
```
Comment (ClementVanPeuter) [inline functions/src/update-stats.ts]: A template string might be more appropriate here. Maybe you could check if a tslint rule exists to enforce their use.

File: functions/src/compute-statistics.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  async (payload: keyPayload) => {
+    if (!config.features.compute_statistics.enabled) {
+      console.info("Compute statistics feature is disabled; aborting");
+      return;
+    }
+    if (payload.key !== adminKey) {
+      console.info("function's payload passed the wrong admin key; aborting");
+      return;
+    }
+    db.collection("votes")
```
Comment (ClementVanPeuter) [inline functions/src/compute-statistics.ts]: :warning: the collection is actually named `vote`

Comment (Ked57) [reply functions/src/compute-statistics.ts]: i can confirm that this is work in progress, i forgot things :(

Comment (hgwood) [reply functions/src/compute-statistics.ts]: Sure! No worries :)

**Comment by Ked57** [comment]
Need help @hgwood 🆘 😉

File: functions/src/firebase-delete-collection.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        await deleteQueryBatch(
+          fireStore,
+          query,
+          batchSize,
+          resolve,
+          reject
+        ).catch(reject);
+      });
+    })
+    .catch(reject);
```
Comment (hgwood) [inline functions/src/firebase-delete-collection.ts]: According to the documentation, this code is a simplified example and is not recommended for production (also it is not good code, quality-wise). The doc instead recommends to use `firebase-tools`. See [here](https://github.com/firebase/firebase-tools/blob/master/lib/firestore/delete.js) for the code and [here](https://github.com/firebase/firebase-tools/blob/master/commands/firestore-delete.js#L78) for how to call it.

File: functions/src/update-stats.ts [hunk 4]
```diff
@@ -12,48 +12,47 @@ interface StatsData {
...
-      return transaction
-        .set(statsCollection.doc(vote.campaign), newCounters)
-        .set(
-          statsCollection
-            .doc(vote.campaign)
-            .collection("votes")
-            .doc(voteId),
-          {}
-        );
+    updateStatsFunction(vote, voteId).catch(e => {
```
Comment (hgwood) [inline functions/src/update-stats.ts]: You should probably look for better names :)

File: firebase-debug.log [hunk 5]
```diff
@@ -0,0 +1,53 @@
+[debug] [2018-10-16T08:31:06.940Z] ----------------------------------------------------------------------
```
Comment (hgwood) [inline firebase-debug.log]: I think this should not be there. Maybe you could add it to .gitignore.

**Pull request merged**